File: Backend/API/User/DTO/ResetPasswordDto.cs
using System.ComponentModel.DataAnnotations;
using Infrastructure.Attributes;

namespace API.User.DTO
{
    public sealed record ResetPasswordDto
    {
        [Required]
        [EmailAddress]
        public string Email { get; init; }

        [Required]
        [Password]
        public string Password { get; init; }

        [Required]
        [Password]
        [Compare(nameof(ConfirmPassword), ErrorMessage = "Passwords not match.")]
        public string ConfirmPassword { get; init; }

        [Required]
        public string Code { get; init; }
    }
}

File: po-rowno-app/src/app/_services/auth/dto/reset-password.dto.ts
export interface ResetPasswordDto {
  /**
   * @format email
   * @minLength 1
   */
  email: string;
  /** @minLength 1 */
  password: string;
  /** @minLength 1 */
  confirmPassword: string;
  /** @minLength 1 */
  code: string;
}
